ccf_transaction_processor/Makefile

```diff
@@ -0,0 +1,136 @@
+# Copyright 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# SCRIPTDIR defaults to the current directory
+# SRCDIR defaults to the parent directory
+# DSTDIR defaults to _dev under the current directory
+# Both can be overridden with environment variables
+
+ifndef PDO_INSTALL_ROOT
+$(error Incomplete configuration, PDO_INSTALL_ROOT is not defined)
+endif
+
+ifndef SGX_MODE
+$(error Incomplete configuration, SGX_MODE is not defined)
+endif
+
+ifdef NINJA_OPTION
+NINJA_J_OPTION = $(NINJA_OPTION)
+else
+NINJA_J_OPTION = 4
+endif
+
+SCRIPTDIR ?= $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
+CCFDIR := $(SCRIPTDIR)/CCF
+DSTDIR ?= $(PDO_INSTALL_ROOT)
+CCFDSTDIR ?= $(PDO_INSTALL_ROOT)/opt/pdo/ccf
+SRCDIR ?= $(abspath $(SCRIPTDIR)/..)
+KEYDIR := $(DSTDIR)/opt/pdo/ccf/keys/
+ETCDIR := $(DSTDIR)/opt/pdo/ccf/etc/
+BINDIR := $(DSTDIR)/opt/pdo/ccf/bin/
+BLDDIR := $(SCRIPTDIR)/build
+
+KEYGEN=$(abspath $(BLDDIR)/keygenerator.sh
+CNFGEN=$(abspath $(PDO_SOURCE_ROOT)/build/__tools__/expand-config)
+
+PYTHON_DIR=$(DSTDIR)/lib/python3.7
+
+# this should be determined by SGX_MODE
+CCF_MODE=virtual
+
+all: environment keys config install
+
+build: build-ccf build-pdo-tp
+
+build-ccf :
+	mkdir -p $(CCFDIR)/build
+	cd $(CCFDIR)/build; cmake -GNinja \
+		-DCOMPILE_TARGETS=virtual \
+		-DCMAKE_INSTALL_PREFIX=${DSTDIR}/opt/pdo/ccf \
+		-DBUILD_END_TO_END_TESTS=OFF \
+		-DBUILD_SMALLBANK=OFF \
+		-DBUILD_TESTS=OFF \
+		-DBUILD_UNIT_TESTS=OFF \
+		..
+	cd $(CCFDIR)/build; ninja -j $(NINJA_J_OPTION); ninja install
```

Comment:
Hmm, you still have the `;` instead of the more robust `&&`.  Note that my patched Makefile i've sent you fixed all of them (as it did remove/cleanup the DSTDIRs ..)

Reply:
this is a small thing and is not how we do it in the PDO Makefile. the && is far more important if there might be a failure in the left side which for these there is no chance of failure. if you have it in a patch.. fine. but there are more important things to fix.

Reply:
patch applied, thank you! (missed it originally, my bad)

Reply:
> there might be a failure in the left side which for these there is no chance of failure.

Not completely true, see the ninja sequence.  In that case, though, what still helps you of course is that if the left (middle) case fails, the right side should presumably fail as well (exception might be if default build calls failing tests which the install probably won't call).

Reply:
that was the one and only place where it was useful and in fact it isn't needed there either if we do the ninja install as the only command. @g2flyer would you please make the pdo build makefile consistent & get a PR that changes the ';' to '&&' there as well?

Reply:
> @g2flyer would you please make the pdo build makefile consistent & get a PR that changes the ';' to '&&' there as well?

Will do.  I'm already working on a PR for sgx 2.10 (need it for FPC and though i also bump PDO, also as there a few left-over inconsistencies), so depending on how big the change and whether i already have to touch makefiles is might piggy-back it (as separate commit) or create a second PR ...